packages/teleport/src/DesktopSession/WarningDropdown.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useRef, useState } from 'react';
+import { Text, Flex, Button, Card, ButtonIcon } from 'design';
+import styled from 'styled-components';
+import { Notification } from 'shared/components/Notification';
+import { Warning, Close } from 'design/Icon';
+import { orange } from 'design/theme/palette';
+import { useClickOutside } from 'shared/hooks/useClickOutside';
+
+import type { NotificationItem } from 'shared/components/Notification';
+
+export function WarningDropdown({ warnings, onRemoveWarning }: Props) {
+  const [showDropdown, setShowDropdown] = useState(false);
+  const ref = useRef(null);
+
+  const toggleDropdown = () => {
+    if (warnings.length > 0) {
+      setShowDropdown(prevState => !prevState);
+    }
+  };
+
+  // Dropdown is always closed if there are no errors to show
+  if (warnings.length === 0 && showDropdown) setShowDropdown(false);
+
+  // Close the dropdown if it's open and the user clicks outside of it
+  const handleClickOutside = (e: MouseEvent) => {
+    if (ref.current && !ref.current.contains(e.target)) {
```

Comment:
The conditional doesn't seem to be necessary now that we're using `useClickOutside`. We can probably just inline this function.